warehouse/query-core/src/main/java/org/apache/commons/jexl2/parser/JexlNodes.java

```diff
@@ -199,4 +202,44 @@ public static ASTNumberLiteral literal(Number n) {
         l.literal = n;
         return l;
     }
+    
+    /**
+     * Remove childToRemove from parent, updating all references on both the parent and childToRemove to be consistent
+     * 
+     * @param parent
+     *            the parent to remove from
+     * @param childToRemove
+     *            the child to remove from parent
+     * @return true if childToRemove was successfully found and removed from parent, false otherwise
+     */
+    public static boolean removeFromParent(JexlNode parent, JexlNode childToRemove) {
```

Comment:
I would love to see this renamed to 'orphanChild'.   I'm 50/50 on whether that's a joke or not.